### PR TITLE
koordlet: refactor resmanager to qosmanager with plug-in framework

### PR DESCRIPTION
**Pull request by zwzhang0107**
### Ⅱ. Does this pull request fix one issue?
#174

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/koordlet/qosmanager/helpers/property.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		resourceQoS = strategy.BEClass
+	default:
+		// should never reach here
+	}
+	return resourceQoS
+}
+
+// getKubeQoSResourceQoSByQoSClass gets pod config by mapping kube qos into koordinator qos.
+// https://koordinator.sh/docs/core-concepts/qos/#koordinator-qos-vs-kubernetes-qos
+func GetKubeQoSResourceQoSByQoSClass(qosClass corev1.PodQOSClass, strategy *slov1alpha1.ResourceQOSStrategy) *slov1alpha1.ResourceQOS {
```
Comment (saintube) [inline pkg/koordlet/qosmanager/helpers/property.go:47]: can we use `extension.GetPodQoSClassWithKubeQoS`

Comment (zwzhang0107) [reply pkg/koordlet/qosmanager/helpers/property.go:47]: it returns *slov1alpha1.ResourceQOS

**Review by stormgbs — COMMENTED** [review]
/LGTM

**Review by hormes — APPROVED** [review]

File: pkg/koordlet/qosmanager/framework/context.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+func (r *Evictor) evictPod(evictPod *corev1.Pod, reason string, message string) bool {
+	podEvictMessage := fmt.Sprintf("evict Pod:%s, reason: %s, message: %v", evictPod.Name, reason, message)
+	_ = audit.V(0).Pod(evictPod.Namespace, evictPod.Name).Reason(reason).Message(message).Do()
+
+	if err := util.EvictPodByVersion(context.TODO(), r.kubeClient, evictPod.Namespace, evictPod.Name, metav1.DeleteOptions{
+		GracePeriodSeconds: nil,
+		Preconditions:      metav1.NewUIDPreconditions(string(evictPod.UID))}, r.evictVersion); err == nil {
+		r.eventRecorder.Eventf(evictPod, corev1.EventTypeWarning, helpers.EvictPodSuccess, podEvictMessage)
+		metrics.RecordPodEviction(evictPod.Namespace, evictPod.Name, reason)
+		klog.Infof("evict pod %v/%v success, reason: %v", evictPod.Namespace, evictPod.Name, reason)
```
[the remark this answers is not in the document]
Comment (zwzhang0107) [reply pkg/koordlet/qosmanager/framework/context.go:90]: maybe later since now it's all start with lowercase.

**Review by stormgbs — APPROVED** [review]
/lgtm

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Pull request merged**